plugins/mumble_plugin.h

```diff
@@ -27,65 +26,30 @@
 # error No MUMBLE_PLUGIN_EXPORT definition available
 #endif
 
-// Visual Studio 2008 x86
-#if _MSC_VER == 1500 && defined(_M_IX86)
-# define MUMBLE_PLUGIN_MAGIC     0xd63ab7ef
-# define MUMBLE_PLUGIN_MAGIC_2   0xd63ab7fe
-# define MUMBLE_PLUGIN_MAGIC_QT  0xd63ab7ee
-// Visual Studio 2010 x86
-#elif _MSC_VER == 1600 && defined(_M_IX86)
-# define MUMBLE_PLUGIN_MAGIC    0xd63ab7f0
-# define MUMBLE_PLUGIN_MAGIC_2  0xd63ab7ff
-# define MUMBLE_PLUGIN_MAGIC_QT 0xd63ab70f
-// Visual Studio 2013 x86
-#elif _MSC_VER == 1800 && defined(_M_IX86)
-# define MUMBLE_PLUGIN_MAGIC    0xd63ab7c0
-# define MUMBLE_PLUGIN_MAGIC_2  0xd63ab7cf
-# define MUMBLE_PLUGIN_MAGIC_QT 0xd63ab7ca
-// Visual Studio 2013 x64
-#elif _MSC_VER == 1800 && defined(_M_X64)
-# define MUMBLE_PLUGIN_MAGIC    0x9f3ed4c0
-# define MUMBLE_PLUGIN_MAGIC_2  0x9f3ed4cf
-# define MUMBLE_PLUGIN_MAGIC_QT 0x9f3ed4ca
-// Visual Studio 2015 x86
-#elif _MSC_VER == 1900 && defined(_M_IX86)
-# define MUMBLE_PLUGIN_MAGIC    0xa9b8c7c0
-# define MUMBLE_PLUGIN_MAGIC_2  0xa9b8c7cf
-# define MUMBLE_PLUGIN_MAGIC_QT 0xa9b8c7ca
-// Visual Studio 2015 x64
-#elif _MSC_VER == 1900 && defined(_M_X64)
-# define MUMBLE_PLUGIN_MAGIC    0x1f2e3dc0
-# define MUMBLE_PLUGIN_MAGIC_2  0x1f2e3dcf
-# define MUMBLE_PLUGIN_MAGIC_QT 0x1f2e3dca
-// Generic plugin magic values for platforms
-// where we do not officially plugins other
-// than "link".
-#else
-# define MUMBLE_PLUGIN_MAGIC    0xf4573570
-# define MUMBLE_PLUGIN_MAGIC_2  0xf457357f
+# define MUMBLE_PLUGIN_MAGIC    0xf4573580
 # define MUMBLE_PLUGIN_MAGIC_QT 0xf457357a
-#endif
 
-#define MUMBLE_PLUGIN_VERSION 2
+#define MUMBLE_PLUGIN_VERSION 3
+
+#define MUMBLE_PID_LOOKUP_EOF  0
+#define MUMBLE_PID_LOOKUP_OK   1
+
+typedef void * MumblePIDLookupContext;
+typedef int MumblePIDLookupStatus;
+typedef MumblePIDLookupStatus (*MumblePIDLookup)(const MumblePIDLookupContext ctx, const wchar_t *str, unsigned long long *pid);
 
 typedef struct _MumblePlugin {
 	unsigned int magic;
-	const std::wstring &description;
-	const std::wstring &shortname;
-	void (MUMBLE_PLUGIN_CALLING_CONVENTION *about)(void *);
-	void (MUMBLE_PLUGIN_CALLING_CONVENTION *config)(void *);
-	int (MUMBLE_PLUGIN_CALLING_CONVENTION *trylock)();
+	unsigned int version;
+	bool retracted;
```

Comment:
Imho this is a bit of overkill for a special case in our plugin auto-updating system. Why have every plugin developer bother to set this flag when only the null plugin will ever set it? Any reason why the magic name could cause problems/was limiting?

Reply:
Actually, no, there aren't any problems by checking if the plugin's name is `Retracted`.

I just think this is a cleaner way of checking if the plugin is retracted.